cli/azd/pkg/input/console.go

```diff
@@ -63,14 +67,35 @@ func (c *AskerConsole) SetWriter(writer io.Writer) {
 
 // Prints out a message to the underlying console write
 func (c *AskerConsole) Message(ctx context.Context, message string) {
-	// Only write to the console during interactive & non-formatted responses.
-	if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
+	if c.formatter != nil && c.formatter.Kind() == output.JsonFormat {
+		// we do not expect formatter.Format to fail on valid input. Discard the error to appease the linter (note
+		// that we implicitly discard the error from fmt.Fprintln later, as well).
+		_ = c.formatter.Format(c.consoleMessageForMessage(message), c.writer, nil)
```

Comment:
Do you think this is too verbose with our output messages? I wonder if we should consider just emitting the most important progress messages for now to minimize verbosity noise. That would require refactoring `Message` vs maybe `InfoMessage`. The key acceptance is to have a good user-experience for VS.

Reply:
> Do you think this is too verbose with our output messages? I wonder if we should consider just emitting the most important progress messages for now to minimize verbosity noise. That would require refactoring Message vs maybe InfoMessage. The key acceptance is to have a good user-experience for VS.

I don't actually feel like this is too verbose today.  In practice, all of these Messages are going to be printed to the console when not run in JSON mode and so we don't end up spewing a bunch of stuff.

The experience for VS today is they would get messages that if just printed as is would result in:

```
Infrastructure provisioning plan completed successfully
Provisioning Azure resources can take some time.

You can view detailed progress in the Azure Portal:
https://portal.azure.com/#blade/HubsExtension/DeploymentDetailsBlade/overview/id/%2Fsubscriptions%2Ffaa080af-c1d8-40ad-9cce-e1a450ca5b57%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Fmatell-azd-json

Created Resource group: rg-matell-azd-json
Created App Service plan: plan-itzzednmw27ng
Created Web App: app-itzzednmw27ng

Azure resource provisioning completed successfully
View the resources created under the resource group rg-matell-azd-json in Azure Portal:
https://portal.azure.com/#@/resource/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/rg-matell-azd-json/overview
```

Which seems like a reasonable amount of output?

Reply:
That's pretty good. I think it does omit key parts like: "Starting infrastructure provisioning plan..." or "Starting Azure resource provisioning...". I"m guessing because those are part of the progress logs.

Reply:
Yes - anything that we emit as a status text that goes in front of the spinner today won't get emitted here.  I think when we want to start emitting that stuff, we'd have a new message like:

```json
{
    "type": "statusMessage",
    "data": {
      "status": "Starting infrastructure provisioning plan."
    }
}
```

And a consumer could decide what to do with these status messages (display the most recent one but when a new one comes in replace the existing status message with the new text).  This is similar in spirit to what the CLI does as it gets progress messages out from the Async operations.